pkg/controller/stack/resolver.go

```diff
@@ -108,3 +117,109 @@ func SearchStack(stackName string, index *Index) ([]Stack, error) {
 
 	return stackRefs, nil
 }
+
+// Returns true if the user specified values Kabanero.Spec.Stacks.Repositories.GitRelease.
+// Note that Kabanero.Spec.Stacks.Repositories.GitRelease.Hostname is excluded because
+// users may or may not specify it when connecting to a public Git repository.
+func isGitReleaseUsable(gitRelease kabanerov1alpha2.GitReleaseSpec) bool {
+	return len(gitRelease.Organization) != 0 && len(gitRelease.Project) != 0 &&
+		len(gitRelease.Release) != 0 && len(gitRelease.AssetName) != 0
+
+}
+
+// Retrieves a stack index file content using HTTP.
+func getStackIndexUsingHttp(repoConf kabanerov1alpha2.RepositoryConfig) ([]byte, error) {
+	url := repoConf.Https.Url
+	// user may specify url to yaml file or directory
+	matched, err := regexp.MatchString(`/([^/]+)[.]yaml$`, url)
+	if err != nil {
+		return nil, err
+	}
+	if !matched {
+		url = url + "/index.yaml"
+	}
+
+	return getFromCache(url, repoConf.Https.SkipCertVerification)
+}
+
+// Retrieves a stack index file content using Git APIs
+func getStackIndexUsingGit(repoConf kabanerov1alpha2.RepositoryConfig) ([]byte, error) {
+	var indexBytes []byte
+
+	// Get a Github client.
+	gclient, err := getGitClient(repoConf.GitRelease)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the release tagged in Github as repoConf.GitRelease.Release.
+	release, response, err := gclient.Repositories.GetReleaseByTag(context.Background(), repoConf.GitRelease.Organization, repoConf.GitRelease.Project, repoConf.GitRelease.Release)
+	if err != nil || response.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("Unable to retrieve object representing Github repository release %v. Configured GitRelease data: %v. Error: %v", repoConf.GitRelease.Release, repoConf.GitRelease, err)
+	}
+	assets := release.Assets
+
+	// Find the asset identified as repoConf.GitRelease.AssetName and download it.
+	for _, asset := range assets {
+		if asset.GetName() == repoConf.GitRelease.AssetName {
+			id := asset.GetID()
+			reader, redirectURL, err := gclient.Repositories.DownloadReleaseAsset(context.Background(), repoConf.GitRelease.Organization, repoConf.GitRelease.Project, id)
```

Comment:
this is fine... but I thought the github client would let you follow re-directs if you supplied an HTTP client to do so with.  Did you try this?
https://github.com/google/go-github/commit/75ddd505ad1c4d7c2d0bd0c5e0ef4e753907fe6a

Reply:
The code is passing an HTTP client. It is the default one, but a client nonetheless. The case that is documented is the one where you get a redirect if the value of the client is nil.

Reply:
I don't see where it's passing a redirect client.  the args are context, org, project, and id.  there should be a 5th arg, which is the client to use for redirects.  unless the code defaults to use the default client if one is not provided (I didn't think that it did).  Is that what is happening here?

Reply:
Ok, i got to the bottom of this. It turns out that we were looking at two different versions of the doc/code. The level of go-github imported by VS code is not the very latest one. The change in go-github that introduced the redirect logic you are talking about made it into a release 3 days ago. The doc you are looking at was, most likely, out there around 25 days ago. So that explains why i HAD to handle the redirects myself and why you thought i didn't need to. I will try to override the go-github version with the very latest and see how it goes.